infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js

```diff
@@ -39,72 +38,117 @@ const web3 = new Web3(keepContractOwnerProvider, null, web3Options)
+      if (grantBalance.gte(grantAmount)) {
+        console.warn(
+          `[${granteeAccount}] requested grant while at limit. Balance: [${grantBalance}].`,
+        )
+        return response.status(400).send(`
+          Token grant failed: your account has the maximum testnet KEEP allowed.\n
+          You can manage your token grants at: https://dashboard.test.keep.network\n
+          If you have questions, you can find us on Discord: https://discord.gg/jqxBU4m\n`
+        )
+      } else {
+        console.log(
+          `Submitting grant for [${grantAmount}] to [${granteeAccount}]...`,
+        )
+        const grantData = Buffer.concat([
+          Buffer.from(granteeAccount.substr(2), 'hex', 20),
+          unlockingDuration.toBuffer('be', 32),
+          start.toBuffer('be', 32),
+          cliff.toBuffer('be', 32),
+          Buffer.from(revocable ? '01' : '00', 'hex'),
+          Buffer.from(permissiveStakingPolicyAddress.substr(2), 'hex', 20)
+        ])
 
-      console.log(
-        `Created grant for ${web3.utils.toBN(
-          grantAmount,
-        )} to: ${granteeAccount}`,
-      )
-      response.send(`
-        Created token grant with ${web3.utils.toBN(
-          grantAmount,
-        )} KEEP for account: ${granteeAccount}
-        You can manage your token grants at: https://dashboard.test.keep.network
-        You can find us on Discord at: https://discord.gg/jqxBU4m\n`)
+        console.log("Test submission...")
+        // Try calling; if this throws, we'll have a proper error message thanks
+        // to handleRevert above.
+        await keepToken.methods
+          .approveAndCall(tokenGrantAddress, grantAmount, grantData)
+          .call({ from: keepContractOwnerAddress })
+
+        console.log("Submitting transaction...")
+        // If the call didn't revert, try submitting the transaction proper.
+        keepToken.methods
+          .approveAndCall(tokenGrantAddress, grantAmount, grantData)
+          .send({ from: keepContractOwnerAddress })
+          .on('transactionHash', (hash) => {
+            console.log(
+              `Submitted grant for [${grantAmount}] to [${granteeAccount}].`,
+              transaction,
+            )
+            response.send(`
+              Created token grant with ${grantAmount} KEEP for account: ${granteeAccount}\n
+              You can follow the transaction at https://ropsten.etherscan.io/tx/${transaction.hash}\n
+              You can manage your token grants at: https://dashboard.test.keep.network .\n
+              You can find us on Discord at: https://discord.gg/jqxBU4m .\n
+            `)
+          })
+          .on('error', (error) => {
+            console.error(
+              `Error with account grant transaction: [${error}]; URL was [${request.url}].`
+            )
+            if (! response.headersSent) {
+              response.status(500).send(`
+                  Token grant failed, try again.\n
```

Comment:
Is there a reason the `500` responses are tabbed more than 2 spaces?

Reply:
That is a good question. Lemme make sure prettier hit this.

Reply:
Ok, prettiered.